lib/promscrape/targetstatus.go

```diff
@@ -104,10 +105,53 @@
 
 func (tsm *targetStatusMap) registerJobNames(jobNames []string) {
 	tsm.mu.Lock()
+	tsm.registerJobsMetrics(jobNames)
 	tsm.jobNames = append(tsm.jobNames[:0], jobNames...)
 	tsm.mu.Unlock()
 }
 
+// registerJobsMetrics registers metrics for new jobs and unregisterMetric metrics for removed jobs
+func (tsm *targetStatusMap) registerJobsMetrics(jobNames []string) {
```

Comment:
Could you name function as `registerJobsMetricsLocked` in order to indicate that function is supposed to be called when `tsm.mu` is locked by caller?

Alternatively, if function could accept `registerJobsMetrics(currentJobNames, newJobNames []string)`. In this case locking mutex won't be required.